File: frontend/lib/domain/model/workout/workout_block_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'workout_block_result.freezed.dart';

@freezed
class WorkoutBlockResult with _$WorkoutBlockResult {
  const factory WorkoutBlockResult({
    required int id,
    required int workoutResultId,
    required int workoutBlockId,
    int? averagePower,
    int? maxPower,
    int? averageCadence,
    required int durationSeconds,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _WorkoutBlockResult;
}

File: backend/app/api/entities/workout_block_result.rb
module Entities
  class WorkoutBlockResult < Grape::Entity
    expose :id
    expose :workout_result_id
    expose :workout_block_id
    expose :average_power
    expose :max_power
    expose :average_cadence
    expose :duration_seconds
    expose :created_at
    expose :updated_at
  end
end
